tests/network/vmi_infosource.go

```diff
@@ -0,0 +1,101 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kvirtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/libvmi"
+	"kubevirt.io/kubevirt/tests/util"
+)
+
+var _ = SIGDescribe("Infosource", func() {
+	var virtClient kubecli.KubevirtClient
+
+	BeforeEach(func() {
+		var err error
+		virtClient, err = kubecli.GetKubevirtClient()
+		Expect(err).NotTo(HaveOccurred(), "Should successfully initialize an API client")
+
+		tests.BeforeTestCleanup()
+	})
+
+	Context("with a vmi and a bridge interface", func() {
+		var vmi *kvirtv1.VirtualMachineInstance
+
+		const originalMac = "02:00:05:05:05:05"
+		const updatedMac = "02:00:b5:b5:b5:b5"
+
+		BeforeEach(func() {
+			bridgeInterface := libvmi.InterfaceDeviceWithBridgeBinding()
+			vmiSpec := libvmi.NewFedora(
+				libvmi.WithInterface(*libvmi.InterfaceWithMac(&bridgeInterface, originalMac)),
+				libvmi.WithNetwork(kvirtv1.DefaultPodNetwork()))
+			tests.AddUserData(vmiSpec, "cloud-init", "#!/bin/bash\nsudo ip link set dev eth0 address "+updatedMac+"\n")
```

Comment:
There is a dedicated build function for this: `WithCloudInitNoCloudUserData`. I did not work for you?